components/places/src/api/places_api.rs

```diff
@@ -110,15 +109,19 @@ pub struct SyncState {
 /// https://github.com/mozilla/uniffi-rs/pull/1063 would fix this, but got some pushback
 /// meaning we are forced into this unfortunate workaround.
 pub fn places_api_new(db_name: impl AsRef<Path>) -> Result<Arc<PlacesApi>> {
-    PlacesApi::new(db_name)
+    let api = PlacesApi::new(db_name)?;
+    let api_weakref: Weak<PlacesApi> = Arc::downgrade(&api);
+    shutdown::register_interrupt(api_weakref);
+    Ok(api)
 }
 
 /// The entry-point to the places API. This object gives access to database
 /// connections and other helpers. It enforces that only 1 write connection
 /// can exist to the database at once.
 pub struct PlacesApi {
     db_name: PathBuf,
-    write_connection: Mutex<Option<PlacesDb>>,
+    pub read_connection: Arc<PlacesConnection>,
+    pub write_connection: Arc<PlacesConnection>,
```

Comment:
places has many read connections. Any reason you can't put interrupt support directly on `PlacesDb`, and "unregister" when dropped?